### DISCUSSION TITLE
How to create scrollable grid layout?

### DISCUSSION THREAD

**Discussion by theredfoxlee** [Textualize/textual#2992] [Q&A] [answered]
Hello! I want to create a grid of static widgets with a size of 20x20 (although the size may change). I would like the grid to be scrollable (to make it readable), but I am unsure how to achieve this.

What I have tried:

1. Using `layout: grid;` with `grid-size: 20 20;` and `overflow: scroll scroll;` does not work as all elements are displayed, even when they have `min-width` and `min-height` set. This makes the text unreadable.
2. When using a single `ScrollableContainer` that contains 20 `Horizontal` containers, each containing 20 `Static`s, the scroll bars do not appear. Even though only part of the grid is presented based on the element size, their height is not respected.

Do you have any ideas how to solve it?

---

Here is an image of what I would like to achieve (assuming that consecutive numbers become visible after scrolling to the right or down):

![image](https://github.com/Textualize/textual/assets/20356922/728fa070-7fc4-43b3-911e-f5355e7a61ca)

**Comment by davep** [comment, marked as answer]
As you've discovered, `grid` is for fitting things within the display, essentially. I think if I was going to do what you're asking here, I'd start out with this sort of framework:

```python
from textual.app import App, ComposeResult
from textual.containers import VerticalScroll, Horizontal
from textual.widgets import Static

class ScrollableGridApp(App[None]):

    CSS = """
    .row {
        height: 5;
    }

    .row Static {
        border: solid cornflowerblue;
        width: 1fr;
        height: 1fr;
    }
    """

    def compose(self) -> ComposeResult:
        with VerticalScroll():
            for y in range(20):
                with Horizontal(classes="row"):
                    for x in range(20):
                        yield Static(f"{y} {x}")

if __name__ == "__main__":
    ScrollableGridApp().run()
```

giving:

![Screenshot 2023-07-23 at 07 59 57](https://github.com/Textualize/textual/assets/28237/7def7b96-870c-4132-89fb-742605465127)

**Reply by davep** [reply]
Was this what you were looking for @theredfoxlee?

**Comment by theredfoxlee** [comment]
@davep - it seems that what I was missing was adding the size to the horizontal container. What you did is exactly what I meant. Thank you for your help!

**Reply by davep** [reply]
Ahh! You wanted scrolling horizontal and vertical? Sorry if I misunderstood the question. Glad you got it sorted.
